pkg/master/master.go

```diff
@@ -360,6 +360,13 @@ func (c completedConfig) New(delegationTarget genericapiserver.DelegationTarget)
 
 	m.GenericAPIServer.AddPostStartHookOrDie("ca-registration", c.ExtraConfig.ClientCARegistrationHook.PostStartHook)
 
+	if m.GenericAPIServer.AuditBackend != nil {
```

Comment:
master.go is kube-apiserver only. This is genericapiserver functionality though and should go there.

Reply:
@sttts There are already many postStart preShutdown hooks registered in master.go. They are the same kind.
https://github.com/kubernetes/kubernetes/blob/master/pkg/master/master.go#L361
https://github.com/kubernetes/kubernetes/blob/master/pkg/master/master.go#L376-L377

Reply:
Because they are needed for kube-apiserver functionality. But auditing is not, it's generic.

Reply:
A little confused, but updated.

Reply:
kube-apiserver is just an instance of a generic apiserver. It has a couple of custom PostStartHooks (some you have linked above). But auditing is a generic concept that can be used by every apiserver, not only by the main kube apiserver. So we need the bootstrapping code in the generic apiserver code path.

Reply:
ok